src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/AddProfessionalDevelopment.vue

```diff
@@ -0,0 +1,425 @@
+        <v-btn
+          size="large"
+          color="primary"
+          :loading="loadingStore.isLoading('application_professionaldevelopment_add_post')"
+          @click="handleAddProfessionalDevelopment"
+        >
+          Save course or workshop
+        </v-btn>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { DateTime } from "luxon";
+import { defineComponent } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import type { VForm, VInput } from "vuetify/components";
+
+import { getApplicationStatus } from "@/api/application";
+import { addProfessionalDevelopment } from "@/api/application";
+import FileUploader from "@/components/FileUploader.vue";
+import { useAlertStore } from "@/store/alert";
+import { useApplicationStore } from "@/store/application";
+import { useCertificationStore } from "@/store/certification";
+import { useLoadingStore } from "@/store/loading";
+import type { Components } from "@/types/openapi";
+import { formatDate } from "@/utils/format";
+import { isNumber } from "@/utils/formInput";
+import * as Rules from "@/utils/formRules";
+
+import type { FileItem } from "./UploadFileItem.vue";
+
+interface ProfessionalDevelopmentData {
+  selection: ("phone" | "email" | "file")[];
+  areAttachedFilesValid: boolean;
+  isFileUploadInProgress: boolean;
+}
+
+export default defineComponent({
+  name: "AddProfessionalDevelopment",
+  components: { FileUploader },
+  props: {
+    applicationId: {
+      type: String,
+      required: true,
+    },
+  },
+  setup: async () => {
+    const applicationStore = useApplicationStore();
+    const alertStore = useAlertStore();
+
+    const loadingStore = useLoadingStore();
+    const router = useRouter();
+    const route = useRoute();
+    const applicationStatus = (await getApplicationStatus(route.params.applicationId.toString()))?.data;
+    const certificationStore = useCertificationStore();
+
+    return {
+      applicationStore,
+      alertStore,
+      loadingStore,
+      router,
+      applicationStatus,
+      certificationStore,
+      formatDate,
+    };
+  },
+  data() {
+    const items = [
+      {
+        title: "Home",
+        disabled: false,
+        href: "/",
+      },
+      {
+        title: "Application",
+        disabled: false,
+        href: `/manage-application/${this.applicationId}`,
+      },
+      {
+        title: "Professional development",
+        disabled: false,
+        href: `/manage-application/${this.applicationId}/professional-development`,
+      },
+      {
+        title: "Add",
+        disabled: true,
+        href: `/manage-application/${this.applicationId}/professional-development/add`,
+      },
+    ];
+
+    const professionalDevelopment: ProfessionalDevelopmentData & Components.Schemas.ProfessionalDevelopment = {
+      selection: [],
+      areAttachedFilesValid: true,
+      isFileUploadInProgress: false,
+    };
```

Comment:
I think we can remove these flags. They were used before to stop the user from submitting, but we moved this logic into the fileUploader component.

Reply:
I need those to show "Uploading in progress" error popup and also to prevent pushing files in the newFiles[] array if the file uploader is in error state.